client/components/data/query-external-contributors/index.tsx

```diff
@@ -0,0 +1,39 @@
+/**
+ * External dependencies
+ */
+import { connect } from 'react-redux';
+import { useEffect, FunctionComponent } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { getExternalContributors } from 'state/sites/external-contributors/actions';
+
+/**
+ * Types
+ */
+import { SiteId } from 'types';
+
+interface OwnProps {
+	siteId: SiteId;
+}
+
+interface DispatchProps {
+	getExternalContributors: ( siteId: SiteId ) => void;
```

Comment:
Pardon me if we've had this conversation already, I don't think we have.

So far I've been recommending against adding types to imported functions like this and instead preferring to let TypeScript infer it. With a full TypeScript implementation it would happen automatically. In our hybrid it gets messy.

That is, we could _either_ leave it as completely inferred, which I have tended to do with `mapStateToProps` and `mapDispatchToProps` _or_ we could be explicit and say `getExternalContributors: typeof getExternalContributors` which isn't really a good option.

In this patch that would mean setting up the `connect` call with an explicit variable…

```ts
const mapDispatchToProps = { getExternalContributors };

const QueryExternalContributors: FunctionComponent< OwnProps & typeof mapDispatchToProps >;
```

Reply:
I think I missed that conversation, but it seems vastly preferable to having multiple over interfaces, or having to change 3 different things to tweak what is added via `connect`.